src/components/transactions/contract.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {useCallback, useState} from 'react';
+
+import {TouchableWithoutFeedback, View} from 'react-native';
+
+import {Color} from '@app/colors';
+import {DataContent, Icon} from '@app/components/ui';
+import {createTheme} from '@app/helpers';
+import {useEffectAsync} from '@app/hooks/use-effect-async';
+import {I18N} from '@app/i18n';
+import {Indexer} from '@app/services/indexer';
+import {OnTransactionRowPress, TransactionListContract} from '@app/types';
+
+export type TransactionPreviewProps = {
+  item: TransactionListContract;
+  onPress: OnTransactionRowPress;
+};
+
+export const TransactionContract = ({
+  item,
+  onPress,
+}: TransactionPreviewProps) => {
+  const [contractName, setContractName] = useState('');
+
+  useEffectAsync(async () => {
+    const name = await Indexer.instance.getContractName(item.to);
```

Comment:
move it on screen layer

Reply:
We should fetch contract name only when we display contract list item. On the screen layer we have renderItem method only